nidirect_breadcrumbs/src/DrivingInstructorsBreadcrumb.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Drupal\nidirect_breadcrumbs;
+
+/**
+ * @file
+ * Generates the breadcrumb trail for driving instructors related pages.
+ */
+
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Breadcrumb builder class for pages/routes relating to driving instructors.
+ *
+ * @package Drupal\nidirect_breadcrumbs
+ */
+class DrivingInstructorsBreadcrumb implements BreadcrumbBuilderInterface {
+
+  /**
+   * Route matches from the service container parameters.
+   *
+   * @var array
+   */
+  protected $routeMatches;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct(array $route_matches, EntityTypeManagerInterface $entity_type_manager) {
+    $this->routeMatches = $route_matches;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('breadcrumb.driving_instructors.matches'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+
+    if (in_array($route_match->getRouteName(), $this->routeMatches)) {
+      $match = TRUE;
+    }
+
+    return $match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(RouteMatchInterface $route_match) {
+    $breadcrumb = new Breadcrumb();
+    $term_storage = $this->entityTypeManager->getStorage('taxonomy_term');
+
+    $links = [];
+    // Home.
+    $links[] = Link::createFromRoute(t('Home'), '<front>');
+    // Motoring term (12).
+    $term = $term_storage->load(12);
```

Comment:
Instead of individually loading, why not use loadMultiple() ?

Reply:
Opted for a very literal approach in the first instance. Latest changes now have a loadMultiple + loop to build up the links.